### DISCUSSION TITLE
Implement an initial set of Vale linter rules

### DISCUSSION THREAD

**Discussion by theletterf** [open-telemetry/opentelemetry.io#3811] [Tooling & Infrastructure]
[Vale](https://vale.sh/) is a prose linter written in Go and distributed as a fast, dependency-free binary for all major operating systems. Compared to classic spell checkers, Vale can use powerful natural language processing [extensions](https://vale.sh/docs/topics/styles/#extension-points) to check for things such as conditional sequences of strings, undefined acronyms, readability, and consistency, to name a few. Vale is currently being used by all major o11y vendors for their docs. You can read more about it in https://passo.uno/posts/first-steps-with-the-vale-prose-linter/

I've extensive prior experience implementing Vale at Splunk (see v[ale-splunk-style-guide](https://github.com/splunk/vale-splunk-style-guide)) and I'd be happy to bring Vale to the OTel docs. I think we could start from an initial set of rules focused on terminology and [Google style guide recommendations](https://github.com/errata-ai/Google) that our current checkers would not be able to catch. We could go for a phased approach where:

1. We would create the rules in a folder, so that anyone could use them if they've Vale installed (optional usage).
2. We could implement a Vale action in GitHub that would check each PR only for the content they bring in ([example](https://github.com/splunk/public-o11y-docs/blob/main/.github/workflows/vale-linter.yml)). Output would be merely informational.
3. After we've tested it for a while, we could integrate the usage of the Vale in red/green pipelines so that critical (Error or Warning) style issues would prevent a merge (think of biased language).

I think there are many benefits to reap from using Vale, especially for maintainers and approvers who want to check contributions against our style guide, especially when using integrations such as the VS Code extension, which checks for style issues in real time, without having to run the check from the terminal:

![image](https://github.com/open-telemetry/opentelemetry.io/assets/1773616/5b8958bd-7cca-4c95-af3e-a071279c3f02)

**Comment by svrnm** [comment]
Thanks for raising this issue. To add what I shared with you 1:1 I am overall in favour of it, but I also have to say that it will not be easy to implement it because (a) we have to replace some existing tooling (textlint especially) and (b) we need to tune vale down to work for us initially and then grow from there and (c) we have to make sure it works well for new contributors as well.

As some historical context: I tried to implement vale in the past, but failed in doing so, that's why I picked textlint and implemented it. 

So, if you can show us a proof of concept and then work towards a working solution (and take care of the maintanance), there is a chance that we can make this change (from textlint&other things to vale)

**Reply by chalin** [reply]
Exploring a PoC some time in the fall might be a good idea.
